src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,592 @@
+    could just return `learningPeriod`. But when some values have fallen out of
+    the historical sliding window of anomaly records, then we have to take those
+    into account as well so we return the `learningPeriod` minus the number
+    shifted out.
+
+    @param numIngested - (int) number of data points that have been added to the
+      sliding window of historical data points.
+    @param windowSize - (int) size of sliding window of historical data points.
+    @param learningPeriod - (int) the number of iterations required for the
+      algorithm to learn the basic patterns in the dataset and for the anomaly
+      score to 'settle down'.
+    **/
+    int diff = numIngested - (int)windowSize;
+    UInt numShiftedOut = max(0, diff);
+    return min(numIngested, max((UInt)0, learningPeriod - numShiftedOut));
+}
+
+
+vector<Real>  updateAnomalyLikelihoods(vector<Real> anomalyScores, UInt verbosity=0) { 
+  /**
+  Compute updated probabilities for anomalyScores using the given params.
+
+  :param anomalyScores: a list of records. Each record is a list with the
+                        following three elements: [timestamp, value, score]
+
+                        Example::
+
+                            [datetime.datetime(2013, 8, 10, 23, 0), 6.0, 1.0]
+
+  :param params: the JSON dict returned by estimateAnomalyLikelihoods
+  :param verbosity: integer controlling extent of printouts for debugging
+  :type verbosity: int
+
+  :returns: 3-tuple consisting of:
+
+            - likelihoods
+
+              numpy array of likelihoods, one for each aggregated point
+
+            - avgRecordList 
+
+              list of averaged input records - stored in runningAverageAnomalies instance member
+
+            - params
+
+              an updated JSON object containing the state of this metric.-- available from historicalScores.getXXX()
+
+  **/
+  if (verbosity > 3) {
+    cout << "In updateAnomalyLikelihoods."<< endl;
+    cout << "Number of anomaly scores: "<<  anomalyScores.size() << endl;
+//    cout << "First 20:", anomalyScores[0:min(20, len(anomalyScores))])
+    cout << "Params: name=" <<  distribution.name << " mean="<<distribution.mean <<" var="<<distribution.variance <<" stdev="<<distribution.stdev <<endl;
+  }
+
+ assert(anomalyScores.size() > 0); // "Must have at least one anomalyScore"
+
+  // Compute moving averages of these new scores using the previous values
+  // as well as likelihood for these scores using the old estimator 
+  vector<Real> likelihoods;
+  for (auto newAverage : runningAverageAnomalies) {
+    likelihoods.push_back(tailProbability(newAverage)); 
+  }
+
+  // Filter the likelihood values. First we prepend the historical likelihoods
+  // to the current set. Then we filter the values.  We peel off the likelihoods
+  // to return and the last windowSize values to store for later.
+  UInt toCrop = min((unsigned long)this->averagedAnomaly.getMaxSize(), runningLikelihoods.size());
```

Comment:
TODO: review, if this should be current  `this->averagedAnomaly.getMaxSize()`, 
or `circular_buffer.capacity()` from this->runningAveragedAnomalies?